fix: ensure non-zero exit code on TestAbortAll (#960)

amytsengtw opened the pull request:
Ensures that Mobly returns a non-zero exit code when `TestAbortAll` is raised, 
preventing CI/CD systems from incorrectly reporting success.

Fixes #960

amytsengtw committed:
fix: ensure non-zero exit code on TestAbortAll (#960)

xpconanfan reviewed (changes requested):
please add a corresponding unit test

xpconanfan commented on mobly/test_runner.py:
```diff
@@ -79,9 +79,10 @@ def main(argv=None):
+        logging.error('Test run aborted by TestAbortAll.')
       except Exception:
-        logging.exception('Exception when executing %s.', config.testbed_name)
+        logging.exception('Exception when executing %s.', config.test_bed_name)
```
I don't think `test_bed_name` exists? why are you changing it?

amytsengtw replied:
Thanks for catching that! It was a typo. I've reverted test_bed_name back to testbed_name in the latest commit

amytsengtw committed:
docs: fix typo in testbed_name

xpconanfan commented on mobly/test_runner.py:
```diff
@@ -79,7 +79,8 @@ def main(argv=None):
         ok = runner.results.is_all_pass and ok
       except signals.TestAbortAll:
-        pass
+        ok = False
```
let's stay consistent with the next block in terms of the order between the two lines

xpconanfan reviewed (changes requested):
Please add a test for this scenario
There are other tests in `test_runner_test` already

amytsengtw committed:
fix: address review comments and add unit test for TestAbortAll exit code

amytsengtw committed:
fix: address review comments and add self-contained unit test

amytsengtw commented:
Fixed the testbed_name typo and added a self-contained unit test test_main_exits_with_error_on_test_abort_all. The test now creates its own temporary config to verify the exit code when signals.TestAbortAll is raised. Ready for another review

xpconanfan commented on tests/mobly/test_runner_test.py:
```diff
@@ -65,6 +65,22 @@ def _assertControllerInfoEqual(self, info, expected_info_dict):
         expected_info_dict['Controller Info'], info.controller_info
     )
 
+  def test_main_exits_with_error_on_test_abort_all(self):
```
move this test next to the other `test_main_*` tests and try to maintain a reasonable order

xpconanfan commented:
Closing per #1006
